### PR TITLE
Transition reference id to data

### PR DESCRIPTION
**Pull request by cscharf**
## Objective
Ensure some future-proofing when needing to expand the reference event information beyond ID. This changes reference id to reference data (json blob) and corrects the associated columns, fields, etc.

### CODE REVIEW

File: src/Core/Models/Api/Request/Accounts/RegisterRequestModel.cs [hunk 1]
```diff
@@ -25,7 +27,7 @@ public class RegisterRequestModel : IValidatableObject
         public Guid? OrganizationUserId { get; set; }
         public KdfType? Kdf { get; set; }
         public int? KdfIterations { get; set; }
-        public string ReferenceId { get; set; }
+        public ReferenceEventData ReferenceData { get; set; }
```
Comment (kspearrin) [inline src/Core/Models/Api/Request/Accounts/RegisterRequestModel.cs:30]: I don't really like coupling data and API models. Even though they are the same in this case. The API interface should always have independent request and response models.

Comment (kspearrin) [reply src/Core/Models/Api/Request/Accounts/RegisterRequestModel.cs:30]: You can just add the API request model for this as a subclass to this model. Or treat reference data more generically as a <string, object> dictionary or something.

File: src/Sql/dbo/Stored Procedures/Organization_Create.sql [hunk 2]
```diff
@@ -27,7 +27,7 @@
     @Gateway TINYINT,
     @GatewayCustomerId VARCHAR(50),
     @GatewaySubscriptionId VARCHAR(50),
-    @ReferenceId VARCHAR(50),
+    @ReferenceData VARCHAR(50),
```
Comment (kspearrin) [inline src/Sql/dbo/Stored Procedures/Organization_Create.sql:30]: Needs to be max right?

File: src/Sql/dbo/Stored Procedures/Organization_Update.sql [hunk 3]
```diff
@@ -27,7 +27,7 @@
     @Gateway TINYINT,
     @GatewayCustomerId VARCHAR(50),
     @GatewaySubscriptionId VARCHAR(50),
-    @ReferenceId VARCHAR(50),
+    @ReferenceData VARCHAR(50),
```
Comment (kspearrin) [inline src/Sql/dbo/Stored Procedures/Organization_Update.sql:30]: Needs to be max right?

File: src/Sql/dbo/Stored Procedures/User_Create.sql [hunk 4]
```diff
@@ -23,7 +23,7 @@
     @Gateway TINYINT,
     @GatewayCustomerId VARCHAR(50),
     @GatewaySubscriptionId VARCHAR(50),
-    @ReferenceId VARCHAR(50),
+    @ReferenceData VARCHAR(50),
```
Comment (kspearrin) [inline src/Sql/dbo/Stored Procedures/User_Create.sql:26]: Needs to be max right?

File: src/Sql/dbo/Stored Procedures/Organization_Create.sql (hunk 2 above)
Comment (cscharf) [reply src/Sql/dbo/Stored Procedures/Organization_Create.sql:30]: Wow, yes, sorry I missed that and good catch, thank you.

**Pull request merged**